wm-system/product/src/main/kotlin/com/ajaxproject/warehouse/infrastructure/adapter/mongo/MongoProductRepository.kt

```diff
@@ -0,0 +1,51 @@
+package com.ajaxproject.warehouse.infrastructure.adapter.mongo
+
+import com.ajaxproject.warehouse.application.port.ProductRepositoryOutPort
+import com.ajaxproject.warehouse.domain.Product
+import com.ajaxproject.warehouse.infrastructure.adapter.mongo.entity.MongoProduct
+import com.ajaxproject.warehouse.infrastructure.adapter.mongo.mapper.mapToDomain
+import com.ajaxproject.warehouse.infrastructure.adapter.mongo.mapper.mapToMongo
+import org.springframework.context.annotation.Primary
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate
+import org.springframework.data.mongodb.core.find
+import org.springframework.data.mongodb.core.findAll
+import org.springframework.data.mongodb.core.findById
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import org.springframework.data.mongodb.core.remove
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+
+@Repository
+@Primary
```

Comment:
redis repository should be primary, not mongo

Reply:
I use mongo as primary because I want to use cache only inside product services and, for example, inside waybill - without cache

Reply:
changed